File: vehicle-rentals-microservices/vehicles-service/src/main/java/com/project/vehicles_service/entity/Vehicle.java
package com.project.vehicles_service.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@AllArgsConstructor
@Data

@Entity
public class Vehicle {
	@Id
	private long vehicleId;
	private String name;
	private String regNo;
	private int wheelCount;
	private String type;
	private String location;
	private String pincode;
	private String description;
	private double pricePerHr;
	private long vendorId;
	private String imagePath;
}

File: onthego/src/app/models/vehicle.ts

export class Vehicle {
    vehicleId: number;
    name: string;
    regNo: string;
    wheelCount: number;
    type: string;
    location: string;
    pincode: string;
    description: string;
    pricePerHr: number;
    vendorId: number;
    imagePath: string;
    reviews:any

    constructor(
        vehicleId: number,
        name: string,
        regNo: string,
        wheelCount: number,
        type: string,
        location: string,
        pincode: string,
        description: string,
        pricePerHr: number,
        vendorId: number,
        imagePath: string
    ) {
        this.vehicleId = vehicleId;
        this.name = name;
        this.regNo = regNo;
        this.wheelCount = wheelCount;
        this.type = type;
        this.location = location;
        this.pincode = pincode;
        this.description = description;
        this.pricePerHr = pricePerHr;
        this.vendorId = vendorId;
        this.imagePath = imagePath;
    }
}
